xml/http_servers.xml

```diff
@@ -957,14 +957,22 @@
   </fingerprint>
 
   <fingerprint pattern="^gunicorn\/([\d.]+)+$">
-    <description>Gunicorn Gunicorn</description>
+    <description>Gunicorn with version</description>
     <example service.version="19.7.1">gunicorn/19.7.1</example>
```

Comment:
Similar version issue with multiple/trailing periods.

`^gunicorn\/(\d+(.\d+)*)+$`

Again, not super crucial.

Reply:
I've modified the version pattern to `(\d+(?:\.\d+)*)` that allows a single version with no period character and also prevents matching version strings that contain multiple period characters in sequence.

```
$ echo "gunicorn/19" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Gunicorn with version", "service.vendor"=>"Gunicorn", "service.product"=>"Gunicorn", "service.version"=>"19", "service.cpe23"=>"cpe:/a:gunicorn:gunicorn:19", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"gunicorn/19"}

$ echo "gunicorn/19.7" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Gunicorn with version", "service.vendor"=>"Gunicorn", "service.product"=>"Gunicorn", "service.version"=>"19.7", "service.cpe23"=>"cpe:/a:gunicorn:gunicorn:19.7", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"gunicorn/19.7"}

$ echo "gunicorn/19.7.1" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Gunicorn with version", "service.vendor"=>"Gunicorn", "service.product"=>"Gunicorn", "service.version"=>"19.7.1", "service.cpe23"=>"cpe:/a:gunicorn:gunicorn:19.7.1", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"gunicorn/19.7.1"}

$ echo "gunicorn/19.7...1" | bundle exec ./bin/recog_match xml/http_servers.xml
FAIL: gunicorn/19.7...1
```